bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -180,11 +182,52 @@ static inline u64 get_on_cpu_cache_miss(u32 *cpu_id)
 	return delta;
 }
 
+static inline void register_new_process_if_not_exist()
+{
+	u64 cgroup_id;
+	u64 pid_tgid = bpf_get_current_pid_tgid();
+	u32 curr_pid = (u32)pid_tgid;
+	u32 curr_tgid = pid_tgid >> 32;
+
+	// create new process metrics
+	struct process_metrics_t *curr_tgid_metrics =
+		bpf_map_lookup_elem(&processes, &curr_tgid);
+	if (!curr_tgid_metrics) {
+		cgroup_id = bpf_get_current_cgroup_id();
+		// the Kernel tgid is the user-space PID, and the Kernel pid is the
+		// user-scape TID
+		process_metrics_t new_process = {
+			.pid = curr_tgid,
+			.cgroup_id = cgroup_id,
+		};
+		bpf_get_current_comm(&new_process.comm, sizeof(new_process.comm));
+		bpf_map_update_elem(
+			&processes, &curr_tgid, &new_process, BPF_NOEXIST);
+
+		// add new thread id (cur_pid) to the precess id (tgid) list
+		bpf_map_update_elem(
+			&pid_tgid_map, &curr_pid, &curr_tgid, BPF_NOEXIST);
```

Comment:
When an eBPF hash map fills up in the kernel, eBPF employs an eviction strategy to handle the insertion of a new key. LRU (Least Recently Used) Eviction is the most common strategy. So, it will remove a old key and replace with a new one.

This should not be a big problem, but we could introduce the logic later to reset this map times to times to avoid trashing problem. However, we should not always reset the map to have a complete list of pids to present loose datapoints. Clean up should be done in a longer interval than the process map reset.

Reply:
You need to use `BPF_MAP_TYPE_LRU_HASH` to have LRU-based eviction.